libraries/Microsoft.Bot.Builder.LanguageGeneration/LGFileLexer.g4

```diff
@@ -1,255 +1,21 @@
-  ;
-
-NEWLINE_IN_BODY
-  : '\r'? '\n' { ignoreWS = true;} -> skip, popMode
-  ;
-
-IF
-  : I F WHITESPACE* ':'  {beginOfTemplateLine}? { ignoreWS = true; beginOfTemplateLine = false;}
-  ;
-
-ELSEIF
-  : E L S E WHITESPACE* I F WHITESPACE* ':' {beginOfTemplateLine}? { ignoreWS = true; beginOfTemplateLine = false;}
-  ;
-
-ELSE
-  : E L S E WHITESPACE* ':' {beginOfTemplateLine}? { ignoreWS = true; beginOfTemplateLine = false;}
-  ;
-
-SWITCH
-  : S W I T C H WHITESPACE* ':' {beginOfTemplateLine}? { ignoreWS = true; beginOfTemplateLine = false;}
-  ;
-
-CASE
-  : C A S E WHITESPACE* ':' {beginOfTemplateLine}? { ignoreWS = true; beginOfTemplateLine = false;}
-  ;
-
-DEFAULT
-  : D E F A U L T WHITESPACE* ':' {beginOfTemplateLine}? { ignoreWS = true; beginOfTemplateLine = false;}
-  ;
-
-ESCAPE_CHARACTER
-  : ESCAPE_CHARACTER_FRAGMENT  { ignoreWS = false; beginOfTemplateLine = false;}
-  ;
-
-EXPRESSION
-  : EXPRESSION_FRAGMENT  { ignoreWS = false; beginOfTemplateLine = false;}
-  ;
-
-TEXT
-  : ~[\r\n]+?  { ignoreWS = false; beginOfTemplateLine = false;}
-  ;
-
-mode MULTILINE_MODE;
-
-MULTILINE_SUFFIX
-  : '```' { inMultiline = false; } -> popMode
-  ;
-
-MULTILINE_ESCAPE_CHARACTER
-  : ESCAPE_CHARACTER_FRAGMENT -> type(ESCAPE_CHARACTER)
-  ;
-
-MULTILINE_EXPRESSION
-  : EXPRESSION_FRAGMENT -> type(EXPRESSION)
-  ;
-
-MULTILINE_TEXT
-  : (('\r'? '\n') | ~[\r\n])+? -> type(TEXT)
-  ;
-
-mode STRUCTURE_NAME_MODE;
-
-WS_IN_STRUCTURE_NAME
-  : WHITESPACE+ -> skip
-  ;
-
-NEWLINE_IN_STRUCTURE_NAME
-  : '\r'? '\n' { ignoreWS = true;} {beginOfStructureProperty = true;}-> skip, pushMode(STRUCTURE_BODY_MODE)
-  ;
-
-STRUCTURE_NAME
-  : (LETTER | NUMBER | '_') (LETTER | NUMBER | '-' | '_' | '.')*
-  ;
-
-TEXT_IN_STRUCTURE_NAME
-  : ~[\r\n]+?
-  ;
-
-mode STRUCTURE_BODY_MODE;
-
-STRUCTURED_COMMENTS
-  : '>' ~[\r\n]* '\r'?'\n' { !inStructuredValue && beginOfStructureProperty}? -> skip
-  ;
-
-WS_IN_STRUCTURE_BODY
-  : WHITESPACE+ {ignoreWS}? -> skip
-  ;
-
-STRUCTURED_NEWLINE
-  : '\r'? '\n' { ignoreWS = true; inStructuredValue = false; beginOfStructureProperty = true;}
-  ;
-
-STRUCTURED_BODY_END
-  : ']' {!inStructuredValue}? { inTemplate = false; beginOfTemplateBody = false;} -> popMode, popMode
-  ;
-
-STRUCTURE_IDENTIFIER
-  : (LETTER | NUMBER | '_') (LETTER | NUMBER | '-' | '_' | '.')* { !inStructuredValue && beginOfStructureProperty}? {beginOfStructureProperty = false;}
-  ;
+NEWLINE : '\r'? '\n';
 
-STRUCTURE_EQUALS
-  : '=' {!inStructuredValue}? {inStructuredValue = true;} 
-  ;
+OPTION : WHITESPACE* '>' WHITESPACE* '!#' ~('\r'|'\n')+ { inTemplate = false; };
 
-STRUCTURE_OR_MARK
-  : '|' { ignoreWS = true; }
-  ;
+COMMENT : WHITESPACE* '>' ~('\r'|'\n')* { !inTemplate }?;
 
-ESCAPE_CHARACTER_IN_STRUCTURE_BODY
-  : ESCAPE_CHARACTER_FRAGMENT { ignoreWS = false; }
-  ;
+IMPORT : WHITESPACE* '[' ~[\r\n[\]]*? ']' '(' ~[\r\n()]*? ')' WHITESPACE* { inTemplate = false;};
```

Comment:
by setting inTemplate = false, are we saying, when facing a import, we consider the template body ends? 

same question to OPTION, which means we support 

```
# hi
>!# @option = value
[import] (b.c)
```
?

Reply:
yes, 
```
# hi
>!# @option = value
[import] (b.c)
```
means template with empty body,  an option and a import definition